### PR TITLE
internal(bigquery): cast time without 'time_since_epoch()'

### PR DESCRIPTION
**Pull request by MeilakhSiarhei**
Based on comments for already merged PR
https://github.com/googleapis/google-cloud-cpp/pull/13103

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13140)

### CODE REVIEW

File: google/cloud/bigquery/v2/minimal/internal/job_request_test.cc [hunk 1]
```diff
@@ -36,23 +36,19 @@ using ::google::cloud::bigquery_v2_minimal_testing::MakePostQueryRequest;
...
-auto static const kMaxInt =
-    std::chrono::duration_cast<std::chrono::milliseconds>(
-        kMax.time_since_epoch())
-        .count();
+auto static const kMinCreationTime = 1111111111111;
+auto static const kMaxCreationTime = 1111111111211;
+std::chrono::system_clock::time_point static const kMinTime{
+    std::chrono::milliseconds{kMinCreationTime}};
+std::chrono::system_clock::time_point static const kMaxTime{
+    std::chrono::milliseconds{kMaxCreationTime}};
```
Comment (devbww) [inline google/cloud/bigquery/v2/minimal/internal/job_request_test.cc:44]: These are reintroducing the dependency on `system_clock`'s epoch.  But `from_time_t()` can be your friend again.
```
auto const kMinTime = std::chrono::system_clock::from_time_t(0) + std::chrono::milliseconds{kMinCreationTimeMs};
```

File: google/cloud/bigquery/v2/minimal/internal/job_request_test.cc [hunk 2]
```diff
@@ -167,8 +163,10 @@ TEST(ListJobsRequestTest, Success) {
       "https://bigquery.googleapis.com/bigquery/v2/projects/1/jobs");
   expected.AddQueryParameter("allUsers", "true");
   expected.AddQueryParameter("maxResults", "10");
-  expected.AddQueryParameter("minCreationTime", std::to_string(kMinInt));
-  expected.AddQueryParameter("maxCreationTime", std::to_string(kMaxInt));
+  expected.AddQueryParameter("minCreationTime",
+                             std::to_string(kMinCreationTime));
+  expected.AddQueryParameter("maxCreationTime",
+                             std::to_string(kMaxCreationTime));
```
Comment (devbww) [inline google/cloud/bigquery/v2/minimal/internal/job_request_test.cc:169]: You can choose, but I think I would recommend just...
```
auto const kMinCreationTime = std::chrono::system_clock::from_time_t(0) + 
                              std::chrono::milliseconds{1111111111111};
auto const kMaxCreationTime = std::chrono::system_clock::from_time_t(0) +
                              std::chrono::milliseconds{1111111111211};
```
and then here using, for example,...
```
    std::to_string(TimePointToUnixMilliseconds(kMinCreationTime))
```

**Review by MeilakhSiarhei — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13140)* status: 0 of 2 files reviewed, 4 unresolved discussions (waiting on @devbww)

___
*[`google/cloud/bigquery/v2/minimal/internal/job_request.cc` line 143 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13140#gh-1395937726:-NjO0CbP0PvR1a1EUJKd:bb6un1j) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/ddc16cac57b14e57cc9ca1fe8af870aa418fa54e/google/cloud/bigquery/v2/minimal/internal/job_request.cc#L143)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

You can remove the comment now.
</blockquote></details>

Done
___
*[`google/cloud/bigquery/v2/minimal/internal/job_request_test.cc` line 40 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13140#gh-1395940484:-NjO0EZ95vz0mR1J8qr7:bb6un1j) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/ddc16cac57b14e57cc9ca1fe8af870aa418fa54e/google/cloud/bigquery/v2/minimal/internal/job_request_test.cc#L40)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

These probably deserve an "Ms" suffix to indicate units.

Also, we're calling them "times", but they're really "durations".

But, see below about getting rid of them altogether.
</blockquote></details>

Done
___
*[`google/cloud/bigquery/v2/minimal/internal/job_request_test.cc` line 44 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13140#gh-1395968735:-NjO0LAO5hLJe_9pHaI0:bb6un1j) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/ddc16cac57b14e57cc9ca1fe8af870aa418fa54e/google/cloud/bigquery/v2/minimal/internal/job_request_test.cc#L44)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

These are reintroducing the dependency on `system_clock`'s epoch.  But `from_time_t()` can be your friend again.
```
auto const kMinTime = std::chrono::system_clock::from_time_t(0) + std::chrono::milliseconds{kMinCreationTimeMs};
```
</blockquote></details>

Done
___
*[`google/cloud/bigquery/v2/minimal/internal/job_request_test.cc` line 169 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13140#gh-1395968842:-NjO0M09CnwgfoPF47yz:bvzfcgq) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/ddc16cac57b14e57cc9ca1fe8af870aa418fa54e/google/cloud/bigquery/v2/minimal/internal/job_request_test.cc#L169)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

You can choose, but I think I would recommend just...
```
auto const kMinCreationTime = std::chrono::system_clock::from_time_t(0) + 
                              std::chrono::milliseconds{1111111111111};
auto const kMaxCreationTime = std::chrono::system_clock::from_time_t(0) +
                              std::chrono::milliseconds{1111111111211};
```
and then here using, for example,...
```
    std::to_string(TimePointToUnixMilliseconds(kMinCreationTime))
```
</blockquote></details>

I need 'time_point' to put into ListJobsRequest object (line #50). And I need 'milliseconds' to compare with the result of the test (line #163). Which means I would need to hardcode '1111111' in 2 places, or use a variable for it. I chose to use the variable.

File: google/cloud/bigquery/v2/minimal/internal/job_request_test.cc [hunk 3]
```diff
@@ -36,12 +36,12 @@ using ::google::cloud::bigquery_v2_minimal_testing::MakePostQueryRequest;
...
-std::chrono::system_clock::time_point static const kMinTime{
-    std::chrono::milliseconds{kMinCreationTime}};
-std::chrono::system_clock::time_point static const kMaxTime{
-    std::chrono::milliseconds{kMaxCreationTime}};
+auto static const kMinCreationTimeMs = 1111111111111;
+auto static const kMaxCreationTimeMs = 1111111111211;
+auto static const kMinTime = std::chrono::system_clock::from_time_t(0) +
+                             std::chrono::milliseconds{kMinCreationTimeMs};
+auto static const kMaxTime = std::chrono::system_clock::from_time_t(0) +
+                             std::chrono::milliseconds{kMaxCreationTimeMs};
```
Comment (devbww) [inline google/cloud/bigquery/v2/minimal/internal/job_request_test.cc:44]: You've now opened up space to call these `kMinCreationTime` and `kMaxCreationTime`.

**Review by devbww — APPROVED** [review]

**Pull request merged**